### ISSUE TITLE
Update node-html-parser to fix a memory leak due to he dependency

### ISSUE THREAD

**Issue by sushantdhiman** [vercel/next.js#28874]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

14.17.0

### What browser are you using?

Firefox

### What operating system are you using?

Arch Linux

### How are you deploying your application?

custom server with docker container

### Describe the Bug

While debugging a memory leak in my application, I discovered `he` module is contributing to memory leak.

Next.js is dependent upon `node-html-parser@1.4.9` which has a memory leak due to `he` module. Upstream discussions

- https://github.com/taoqf/node-html-parser/issues/106
- https://github.com/mathiasbynens/he/issues/82

### Expected Behavior

Next.js should update its `node-html-parser` dependency to latest `4.1.4`, so this memory leak should not affect Next.js applications

### To Reproduce

1. Start your application with `--inspect`mode.
2. Take heap snapshot
2. Browser some pages (Not sure if this is related to `next/image`, but I was debugging a memory leak for that)
3. Take a new heap snapshot, compare with first snapshot

![Leak](https://user-images.githubusercontent.com/9989487/132334545-40d30fde-a8ee-47e7-83d7-941d2f2355e4.png)

**Comment by huozhi** [comment]
Hi @sushantdhiman are you using `next build` and `next start` to start the application?
Did you see any decreasing after the 2rd time (the 3rd, 4th) snapshots?

**Comment by sushantdhiman** [comment]
Hi @huozhi

I am using `next build` then `node --experimental-modules server/index.js` (custom express server). Memory doesn't decrease for next snapshots.

Main memory leak in my application appears when I try to use any pages (SSR) that use `next/image`(using sharp), It seems memory references are never released by image optimization process. But I can't pin point where the issue is, I am still investigating.

During my investigation I discovered `he` module from `node-html-parser` is also leaking (but not the main culprit). I found some discussions related to memory leak in `node-html-parser`, so I am reporting that.

For now just updating `node-html-parser` is enough to close this issue. If I discovered real problem with `next/image` I will create a new issue or report here.
